Marlin/src/gcode/feature/mixing/M163-M165.cpp

```diff
@@ -30,38 +30,33 @@
 /**
  * M163: Set a single mix factor for a mixing extruder
  *       This is called "weight" by some systems.
+ *       The 'P' values must sum to 1.0 or must be followed by M164 to normalize them.
  *
  *   S[index]   The channel index to set
  *   P[float]   The mix value
- *
  */
 void GcodeSuite::M163() {
   const int mix_index = parser.intval('S');
-  if (mix_index < MIXING_STEPPERS) {
-    float mix_value = parser.floatval('P');
-    NOLESS(mix_value, 0.0);
-    mixing_factor[mix_index] = RECIPROCAL(mix_value);
-  }
+  if (mix_index < MIXING_STEPPERS)
+    mixing_factor[mix_index] = MIN(parser.floatval('P'), 0.0);
```

Comment:
Now you mess up the relation when your input is above 1. Second example in the wiki.

Reply:
Already patched to use `MAX` — typing fast here!

The point is to allow any values to be given, which only relate to one another as a group. Hence, `M164` is absolutely required if there are factors given in proportions (values over 1.0). `M163` absolutely requires followup by `M164` if the given factors don't already sum to 1.0.

Reply:
Already patched to use `MAX` — typing fast here!

The point is to allow **any values** to be given, which **only relate to one another as a group**. Hence, `M164` is **absolutely required** if there are factors given that are over 1.0 or otherwise don't sum to 1.0.

To reiterate (and I'll add this to the documentation) `M163` _**must**_ be followed up by `M164` if the given factors don't already sum to 1.0.

Reply:
The examples are 100% valid. `M164` normalizes the mix.

Reply:
I hope now it is more clear that the reason for `RECIPROCAL` in `M163` was to provide the value that the old Stepper / Planner code needed to **scale up** the Bresenham **ceiling** for the mixing stepper counters with the old (1.1.8) code. The `RECIPROCAL` there should have simply been **removed** when the refactor was done, because the mix factors are now used to **scale down** the E **addends** for the mixing steppers instead.

Reply:
Sorry!

Reply:
No worries. For some reason the things that confuse me the most always seem to involve reciprocals.